storage/storage.go

```diff
@@ -71,6 +73,40 @@ func (s *impl) GetUnseen(logID int64) ([]SubmittedCert, error) {
 	return certs, nil
 }
 
+// GetRandSeen returns a random certificate that has been marked seen
+func (s *impl) GetRandSeen(logID int64) (*SubmittedCert, error) {
+	rows, err := s.db.Query("SELECT ID FROM SubmittedCerts WHERE LogID = ? and Seen IS NOT NULL ORDER BY Timestamp ASC LIMIT 1000", logID)
+	if err != nil {
+		return nil, err
+	}
+	var ids []int
+	for rows.Next() {
+		var id int
+		if err = rows.Scan(&id); err != nil {
+			_ = rows.Close()
+			return nil, err
+		}
+		ids = append(ids, id)
+	}
+	_ = rows.Close()
+
+	// randomly pick an id from the list
+	id := ids[rand.Intn(len(ids))]
```

Comment:
I think you can accomplish this with 1 query and less Go by doing:

`SELECT ID, Cert, SCT, Timestamp FROM SubmittedCerts WHERE LogID = ? AND Seen IS NOT NULL ORDER BY RANDOM() LIMIT 1;`

Reply:
So this was my first approach, but was mainly dissuaded due to the somewhat 'magic' aspect, and also because sqlite loads all relevant rows into memory before doing the sort, so as the table grows this will get more and more expensive (and likely eat all memory on even a modestly sized box quite quickly).

I think there might be a less expensive approach though by doing `SELECT ID, Cert, SCT, Timestamp FROM SubmittedCerts WHERE LogID = ? AND Seen IS NOT NULL AND ID = (SELECT id FROM SubmittedCerts ORDER BY RANDOM() LIMIT 1);`. That said it's still likely to show reduced performance/higher memory usage over time.